sympy/polys/matrices/tests/test_domainmatrix.py

```diff
@@ -1360,3 +1360,211 @@ def test_DomainMatrix_pickling():
     assert pickle.loads(pickle.dumps(dM)) == dM
     dM = DomainMatrix([[ZZ(1), ZZ(2)], [ZZ(3), ZZ(4)]], (2, 2), ZZ)
     assert pickle.loads(pickle.dumps(dM)) == dM
+
+
+def test_DomainMatrix_fflu():
+    A = DomainMatrix([[ZZ(1), ZZ(2)], [ZZ(3), ZZ(4)]], (2, 2), ZZ)
+    P, L, D, U = A.fflu()
+    assert P.shape == A.shape
+    assert L.shape == A.shape
+    assert D.shape == A.shape
+    assert U.shape == A.shape
+    assert L[0, 1] == DomainScalar(ZZ(0), ZZ)
+    assert U[1, 0] == DomainScalar(ZZ(0), ZZ)
+    assert D[0, 1] == DomainScalar(ZZ(0), ZZ)
+    assert D[1, 0] == DomainScalar(ZZ(0), ZZ)
```

Comment:
It would be better to have something here like:
```
assert L == DM([[1, 2], [3, 4], ZZ)
```
Generally `DM` could be used here rather than other constructors.